base/compiler/ssair/EscapeAnalysis/EscapeAnalysis.jl

```diff
@@ -0,0 +1,2027 @@
+baremodule EscapeAnalysis
+
+export
+    analyze_escapes,
+    getaliases,
+    isaliased,
+    has_no_escape,
+    has_arg_escape,
+    has_return_escape,
+    has_thrown_escape,
+    has_all_escape
+
+const _TOP_MOD = ccall(:jl_base_relative_to, Any, (Any,), EscapeAnalysis)::Module
+
+# imports
+import ._TOP_MOD: ==, getindex, setindex!
+# usings
+import Core:
+    MethodInstance, Const, Argument, SSAValue, PiNode, PhiNode, UpsilonNode, PhiCNode,
+    ReturnNode, GotoNode, GotoIfNot, SimpleVector, MethodMatch, CodeInstance,
+    sizeof, ifelse, arrayset, arrayref, arraysize
+import ._TOP_MOD:     # Base definitions
+    @__MODULE__, @eval, @assert, @specialize, @nospecialize, @inbounds, @inline, @noinline,
+    @label, @goto, !, !==, !=, ≠, +, -, *, ≤, <, ≥, >, &, |, include, error, missing, copy,
+    Vector, BitSet, IdDict, IdSet, UnitRange, Csize_t, Callable, ∪, ⊆, ∩, :, ∈, ∉,
+    in, length, get, first, last, haskey, keys, get!, isempty, isassigned,
+    pop!, push!, pushfirst!, empty!, delete!, max, min, enumerate, unwrap_unionall,
+    ismutabletype
+import Core.Compiler: # Core.Compiler specific definitions
+    Bottom, InferenceResult, IRCode, Instruction, Signature,
+    MethodResultPure, MethodMatchInfo, UnionSplitInfo, ConstCallInfo, InvokeCallInfo,
+    isbitstype, isexpr, is_meta_expr_head, println, widenconst, argextype, singleton_type,
+    fieldcount_noerror, try_compute_field, try_compute_fieldidx, hasintersect, ⊑,
+    intrinsic_nothrow, array_builtin_common_typecheck, arrayset_typecheck,
+    setfield!_nothrow, alloc_array_ndims, call_sig, argtypes_to_type, is_builtin,
+    is_return_type, istopfunction, validate_sparams, specialize_method, stmt_effect_free,
+    check_effect_free!, invoke_rewrite, IR_FLAG_EFFECT_FREE
+
+if _TOP_MOD === Core.Compiler
+    include(@__MODULE__, "compiler/ssair/EscapeAnalysis/disjoint_set.jl")
```

Comment:
Revise doesn't seem to like this line

```
ianatol@arctic1:~/julia-ea/julia$ ./julia 
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.8.0-DEV.1465 (2022-02-08)
 _/ |\__'_|_|_|\__'_|  |  avi/EscapeAnalysis/7b3e664c08* (fork: 7 commits, 1 day)
|__/                   |

(@v1.8) pkg> activate .
  Activating project at `~/julia-ea/julia`

julia> using Revise

julia> Revise.track(Core.Compiler)
ERROR: MethodError: Cannot `convert` an object of type Module to an object of type String
Closest candidates are:
  convert(::Type{String}, ::String) at ~/julia-ea/julia/usr/share/julia/base/essentials.jl:220
  convert(::Type{T}, ::T) where T<:AbstractString at ~/julia-ea/julia/usr/share/julia/base/strings/basic.jl:231
  convert(::Type{T}, ::AbstractString) where T<:AbstractString at ~/julia-ea/julia/usr/share/julia/base/strings/basic.jl:232
  ...
Stacktrace:
 [1] top-level scope
   @ base/compiler/ssair/EscapeAnalysis/EscapeAnalysis.jl:40
Revise evaluation error at /home/ianatol/julia-ea/julia/base/compiler/ssair/EscapeAnalysis/EscapeAnalysis.jl:40
```

Reply:
yeah maybe Revise doesn't reason about 2-arg form `include` (which is uncommon in usual package development).
We may want to fix on Revise side.

Reply:
alright, Revise should be functional on EA https://github.com/aviatesk/EscapeAnalysis.jl/commit/43cdad0ead1b446eaaf44b13c90876b69d2bae58